modules/kubeconfig/main.tf

```diff
@@ -36,12 +28,28 @@ data "template_file" "kubeconfig_sa" {
     cluster_name = var.cluster_name
     endpoint     = var.endpoint
     name         = local.service_account_name
-    ca_crt       = base64encode(lookup(data.kubernetes_secret.sa_secret.0.data,"ca.crt", ""))
-    token        = lookup(data.kubernetes_secret.sa_secret.0.data,"token", "")
+    ca_crt       = base64encode(lookup(kubernetes_secret.sa_secret.0.data,"ca.crt", ""))
+    token        = lookup(kubernetes_secret.sa_secret.0.data,"token", "")
     namespace    = var.namespace
   }
 }
 
+# 1.24 change: Create service account secret
+resource "kubernetes_secret" "sa_secret" {
+  count = var.create_static_kubeconfig ? 1 : 0
```

Comment:
thanks for adjusting this one. But now What if the user does not request a static kube config file and its generated by the provider and that provider is using 1.24. Does that cause issues in the cluster? or do we still need to create a default secret for the cluster? Just verifying.

Reply:
Looking at the code for provider based kubeconfig the values are added using the data source element at line #8 with the variables defined on line #15-#18. Value of these variables is probably generated here: https://github.com/sassoftware/viya4-iac-azure/blob/main/main.tf#L169
This is independent of the service_account secret.

Reply:
I understand we don't create the secret, but does the provider create a default secret the cluster needs if our code is not being used to generate the kubeconfig file? Wondering if the SAS deployment is dependent on this and if there's a need for a a default secret in the entire cluster. The issue I believe is larger than the scope of what might be captured here.